_rules/explicit-SVG-image-has-name-7d6734.md

```diff
@@ -0,0 +1,187 @@
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `<svg>` element has an explicit role of `img` and an accessible name from the `<title>` element that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<title>1 circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed Example 2
+
+The `<circle>` element has an explicit role of `graphics-symbol` and an accessible name from the `aria-label` attribute that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="https://www.w3.org/2000/svg" role="img">
+	<circle
+		role="graphics-symbol"
+		cx="50"
+		cy="50"
+		r="40"
+		stroke="green"
+		stroke-width="4"
+		fill="yellow"
+		aria-label="1 circle"
+	></circle>
+</svg>
+```
+
+#### Passed Example 3
+
+The `<svg>` element has an explicit role of `graphics-document` and an accessible name from the `<title>` element that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document" width="100" height="100">
+	<title>1 circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `<svg>` element has an explicit role of `img` but has no accessible name.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed Example 2
+
+The `<svg>` element has an explicit role of `img` but has only whitespace in the `<title>` element for the accessible name.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img">
+	<title> </title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed Example 3
+
+The `<svg>` element has an explicit role of `img`, is included in the accessibility tree, but the `<title>` element is empty.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img">
+	<title></title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed Example 4
+
+The `<circle>` element has an explicit role of `graphics-symbol` but does not have an accessible name.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg">
+	<circle role="graphics-symbol" cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
```

Comment:
This is (also) inapplicable because there is no explicit role (same as Inapplicable Example 2). This should be changed or the example should be removed.